Sources/MongoSwift/MongoClient.swift

```diff
@@ -129,7 +134,7 @@ public class MongoClient {
 
         self._client = mongoc_client_new_from_uri(uri)
         guard self._client != nil else {
-            throw MongoError.invalidClient()
+            throw UserError.invalidArgumentError(message: "libmongoc not built with TLS support.")
```

Comment:
is this the only possible reason `mongoc_client_new_from_uri` can fail?

Reply:
Yep. The relevant libmongoc code:

```
mongoc_client_t *
_mongoc_client_new_from_uri (mongoc_topology_t *topology)
{
   BSON_ASSERT (topology);

#ifndef MONGOC_ENABLE_SSL
    if (mongoc_uri_get_ssl (topology->uri)) {
      MONGOC_ERROR ("Can't create SSL client, SSL not enabled in this build.");
      return NULL;
   }
#endif

   client = (mongoc_client_t *) bson_malloc0 (sizeof *client);
```
So if topology (which is created in `mongoc_client_new_from_uri`, which wraps the function above) is null or the malloc fails, the program will crash. Only null return is the TLS case.